plugins/outputs/cloudwatch/cloudwatch.go

```diff
@@ -523,29 +525,35 @@ func (c *CloudWatch) ProcessRollup(rawDimension []*cloudwatch.Dimension) [][]*cl
 			}
 			i += 1
 		}
-		if i == len(targetDimensions) && !reflect.DeepEqual(rawDimension, extraDimensions) {
+		// skip if same number of dimensions as the original metric
+		if i == len(targetDimensions) && len(targetDimensions) != len(rawDimensions) {
```

Comment:
We are checking `len(targetDimensions) != len(rawDimensions)` after the inner loop.
We could have checked it before the inner loop and saved some time.

The inner loop is small and probably not that expensive so I don't care either way.

Reply:
You're right. Will make the change.